src/serve_session.rs

```diff
@@ -118,6 +118,9 @@ impl ServeSession {
                 });
             }
         };
+        if root_project.name.is_none() {
+            return Err(ServeSessionError::NoProjectName);
+        }
```

Comment:
Does this PR actually close #858? My understanding of the issue is that this should be allowed, and the project name in this case should match the name of the parent folder. I could be misunderstanding...

Reply:
That... is a concern actually. I was thinking that it was intended to be used in packages, so they'd never be the top-level project file but upon rereading the issue it's clear that I misunderstood.

Simple enough to fix though.